lib/Model/DlibCnnHogModel/DlibCnnHogModel.php

```diff
@@ -0,0 +1,193 @@
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\FaceRecognition\Model\DlibCnnHogModel;
+
+use OCP\IDBConnection;
+
+use OCA\FaceRecognition\Helper\FaceRect;
+
+use OCA\FaceRecognition\Model\IModel;
+
+use OCA\FaceRecognition\Model\DlibCnnModel\DlibCnn5Model;
+use OCA\FaceRecognition\Model\DlibHogModel\DlibHogModel;
+
+class DlibCnnHogModel implements IModel {
+
+	/*
+	 * Model files.
+	 */
+	const FACE_MODEL_ID = 4;
+	const FACE_MODEL_NAME = "CnnHog5";
+	const FACE_MODEL_DESC = "Default Cnn model with Hog validation, and 5 point landmarks preprictor";
+	const FACE_MODEL_DOC = "";
+
+	/** @var IDBConnection */
+	private $connection;
+
+	/** @var DlibCnn5Model */
+	private $dlibCnn5Model;
+
+	/** @var DlibHogModel */
+	private $dlibHogModel;
+
+	/**
+	 * DlibCnnHogModel __construct.
+	 *
+	 * @param IDBConnection $connection
+	 * @param DlibCnn5Model $dlibCnn5Model
+	 * @param DlibHogModel $dlibHogModel
+	 */
+	public function __construct(IDBConnection   $connection,
+	                            DlibCnn5Model   $dlibCnn5Model,
+	                            DlibHogModel    $dlibHogModel)
+	{
+		$this->connection       = $connection;
+		$this->dlibCnn5Model    = $dlibCnn5Model;
+		$this->dlibHogModel     = $dlibHogModel;
+	}
+
+	public function getId(): int {
+		return static::FACE_MODEL_ID;
+	}
+
+	public function getName(): string {
+		return static::FACE_MODEL_NAME;
+	}
+
+	public function getDescription(): string {
+		return static::FACE_MODEL_DESC;
+	}
+
+	public function getDocumentation(): string {
+		return static::FACE_MODEL_DOC;
+	}
+
+	public function isInstalled(): bool {
+		if (!$this->dlibCnn5Model->isInstalled())
+			return false;
+		if (!$this->dlibHogModel->isInstalled())
+			return false;
+		return true;
+	}
+
+	public function meetDependencies(string &$error_message): bool {
+		if (!$this->dlibCnn5Model->meetDependencies($error_message)) {
+			$error_message .= " This Model depend on Model 1 and must install it.";
+			return false;
+		}
+		if (!$this->dlibHogModel->meetDependencies($error_message)) {
+			$error_message .= " This Model depend on Model 3 and must install it.";
+			return false;
+		}
+		return true;
+	}
+
+	public function getMaximumArea(): int {
+		return $this->dlibCnn5Model->getMaximumArea();
+	}
+
+	public function getPreferredMimeType(): string {
+		return $this->dlibCnn5Model->getPreferredMimeType();
+	}
+
+	public function install() {
+		if ($this->isInstalled()) {
+			return;
+		}
+
+		// Insert on database and enable it
```

Comment:
This is out of any objective of this PR, but it is something I have been thinking about.
The `facerecog_models` table is currently completely useless. I have my doubts for some future development, but now it could be removed without problems, since all the queries are made with Id configured in the SettingsService.

Reply:
I guess...why not. I see its value as a way to have "foreign keys" and maybe in cases with complex upgrades/downgrades/custom models. But even any of this might not be an issue. Other than that, I am fine if you want to throw it away:)